heartbeat/ocivip

```diff
@@ -0,0 +1,238 @@
+
+<parameter name="ocicli" unique="0">
+<longdesc lang="en">
+OCI Command line interface (CLI) tools
+</longdesc>
+<shortdesc lang="en">OCI cli tools</shortdesc>
+<content type="string" default="${OCF_RESKEY_ocicli_default}" />
+</parameter>
+
+<parameter name="secondary_private_ip" unique="1" required="1">
+<longdesc lang="en">
+reserved secondary private ip for compute instance
+</longdesc>
+<shortdesc lang="en">reserved secondary private ip for compute instance</shortdesc>
+<content type="string" default="" />
+</parameter>
+
+<parameter name="api_delay" unique="0">
+<longdesc lang="en">
+a short delay between API calls, to avoid sending API too quick
+</longdesc>
+<shortdesc lang="en">a short delay between API calls</shortdesc>
+<content type="integer" default="${OCF_RESKEY_api_delay_default}" />
+</parameter>
+
+</parameters>
+
+<actions>
+<action name="start"        timeout="30s" />
+<action name="stop"         timeout="30s" />
+<action name="monitor"      timeout="30s" interval="20s" depth="0" />
+<action name="migrate_to"   timeout="30s" />
+<action name="migrate_from" timeout="30s" />
+<action name="meta-data"    timeout="5s" />
+<action name="validate"     timeout="10s" />
+<action name="validate-all" timeout="10s" />
+</actions>
+</resource-agent>
+END
+}
+
+#######################################################################
+
+ocivip_usage() {
+    cat <<END
+usage: $0 {start|stop|monitor|migrate_to|migrate_from|validate|validate-all|meta-data}
+
+Expects to have a fully populated OCF RA-compliant environment set.
+END
+}
+
+ocivip_start() {
+    ocivip_monitor && return $OCF_SUCCESS
+
+    $OCICLI network vnic assign-private-ip --vnic-id $VNIC_ID \
+        --unassign-if-already-assigned \
+        --ip-address ${SECONDARY_PRIVATE_IP}
+    ip addr add ${SECONDARY_PRIVATE_IP}/24 dev ${PRIMARY_IFACE} label ${PRIMARY_IFACE}:0
+    RET=$?
+
+    # delay to avoid sending request too fast
+    sleep ${OCF_RESKEY_api_delay}
+
+    if [ $RET -ne 0 ]; then
+        return $OCF_NOT_RUNNING
+    fi
+
+    ocf_log info "secondary_private_ip has been successfully brought up (${SECONDARY_PRIVATE_IP})"
+    return $OCF_SUCCESS
+}
+
+ocivip_stop() {
+    ocivip_monitor || return $OCF_SUCCESS
+
+    $OCICLI network vnic unassign-private-ip --vnic-id $VNIC_ID \
+        --ip-address ${SECONDARY_PRIVATE_IP}
+    ip addr del ${SECONDARY_PRIVATE_IP}/24 dev ${PRIMARY_IFACE}:0
+    RET=$?
+
+    # delay to avoid sending request too fast
+    sleep ${OCF_RESKEY_api_delay}
+
+    if [ $RET -ne 0 ]; then
+        return $OCF_NOT_RUNNING
+    fi
+
+    ocf_log info "secondary_private_ip has been successfully brought down (${SECONDARY_PRIVATE_IP})"
+    return $OCF_SUCCESS
+}
+
+ocivip_monitor() {
+    $OCICLI network private-ip list --vnic-id $VNIC_ID | grep -q "${SECONDARY_PRIVATE_IP}"
+
+    RET=$?
+
+    if [ $RET -ne 0 ]; then
+        return $OCF_NOT_RUNNING
+    fi
+    return $OCF_SUCCESS
+}
+
+ocivip_validate() {
+    check_binary ${OCICLI}
```

Comment:
You should also check for `jq` here.

Reply:
Fixed [here](https://github.com/ClusterLabs/resource-agents/pull/1751/commits/a76312ed28a0f9ce3ed19a7bf111443d5ba2f5e1)